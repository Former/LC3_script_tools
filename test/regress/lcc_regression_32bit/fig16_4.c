#include <stdio.h>

void NewSwap(int *firstVal, int *secondVal);

int main()
{
  int valueA = 312345;
  int valueB = 454321;

  printf("Before Swap ");
  printf("valueA = %d and valueB = %d\n", valueA, valueB);

  NewSwap(&valueA, &valueB);

  printf("After  Swap ");
  printf("valueA = %d and valueB = %d\n", valueA, valueB);
  return 0;
}

void NewSwap(int *firstVal, int *secondVal)
{
  int tempVal;          /* Holds firstVal when swapping */

  tempVal = *firstVal;
  *firstVal = *secondVal;
  *secondVal = tempVal;
}
