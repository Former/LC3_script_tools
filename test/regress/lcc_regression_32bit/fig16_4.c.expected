Before Swap valueA = 312345 and valueB = 454321
After  Swap valueA = 454321 and valueB = 312345
